packages/wdio-config/src/constants.ts

```diff
@@ -88,3 +88,18 @@ export const SUPPORTED_HOOKS: (keyof Services.Hooks)[] = [
 export const SUPPORTED_FILE_EXTENSIONS = [
     '.js', '.mjs', '.es6', '.ts', '.feature', '.coffee', '.cjs'
 ]
+
+export enum ConfigEnvironmentContext {
+    Unknown = 'unknown',
+    Launcher = 'launcher',
+    Runner = 'runner'
+}
+
+export const AUTOCOMPILE_CONFIG_FILENAME = '.wdiocompile'
```

Comment:
I don't think such a file is needed. If we can compile the config in the runner process and only need the compiler options in the worker we can just pass along the options from the runner process into the worker. Here is how this would go:

- the CLI launcher reads the config initially (https://github.com/webdriverio/webdriverio/blob/main/packages/wdio-cli/src/launcher.ts#L58-L60), we can allow users to pass in some CLI args as you have implemented it already to allow modify compile options for that part. My assumption tho is that we don't need them normally there so compiling the config file should be fine
- we can now access the compile options defined in the config and pass them into the worker as [args](https://github.com/webdriverio/webdriverio/blob/main/packages/wdio-cli/src/launcher.ts#L377)
- these args are then available in the [runner](https://github.com/webdriverio/webdriverio/blob/main/packages/wdio-runner/src/index.ts#L86) and we can pass the compile options along when we [read the config file](https://github.com/webdriverio/webdriverio/blob/main/packages/wdio-runner/src/index.ts#L95)

That should make this a bit easer, what do you think?

Reply:
This would also make the `ConfigEnvironmentContext` obsolete.

Reply:
Good call. Are you sure we need to pass along through the args since it re-loads the file anyway? I have it currently doing that and it seems to work fine, but I'm totally open to what you are asking just curious if we even need to at this point :)

Reply:
Running into a problem with the config lifecycle where the current autocompile happens during `addConfigFile` calls and the args come in through `merge` calls which currently always happen sequentially after addConfigFile calls. These only seem to be separate because the runner shuts down if addConfigFile throws.

If we could combine them then we could check to see if the args has a autoCompileOpts and use that in the same way `ConfigEnvironmentContext` was being used to prevent double auto-compiler loads.

The reason `ConfigEnvironmentContext` worked before was it allowed addConfigFile to run ts-node ASAP in the launcher and to not run at all in the runner.

Merging them would allow us to have only one autoCompile check and call, where with `ConfigEnvironmentContext` before I had two of them, but it changes the API surface around that try/catch a little.

TL;DR; addConfigFile wants to run ts-node ASAP, so without altering the code there is no way to prevent a double-load in the runner via the args because addConfigFile will have already loaded ts-node with defaults by the time the merge code could load it using the args. :(

Reply:
Not merging them is also an option, but would require adding a flag if not to ConfigParser to the function calls, effectively bringing back `ConfigEnvironmentContext` just maybe in another name :/

Reply:
Ah, I am moving towards a solution of exposing a third function on ConfigParser to trigger auto-compilation, and triggering it before feeding files or merging in the launcher, and triggering it after those steps in the runner. Seems to do the trick!